docs/user-guide/nginx-configuration/configmap.md

```diff
@@ -1246,6 +1247,10 @@ Sets the port of syslog server. _**default:**_ 514
 A comma-separated list of locations on which http requests will never get redirected to their https counterpart.
 _**default:**_ "/.well-known/acme-challenge"
 
+## global-allowed-response-headers
+
+A comma-separated list of allowed response headers inside the [custom headers annotations](https://github.com/kubernetes/ingress-nginx/blob/main/docs/user-guide/nginx-configuration/annotations.md#custom-headers)
```

Comment:
Would it be possible document how to allow any arbitrary headers (for example for an isolated "dev" environment)?

Reply:
@rikatz do you have a preference?

Reply:
> or add a special bool to enable the function.

I like this. Its behavior is controlled by clear switches. 
Allowing arbitrary headers is a dangerous action.